internal/webtest/webtest.go

```diff
@@ -0,0 +1,78 @@
+// Copyright 2016 Google Inc. All rights reserved.
+// Use of this source code is governed by the Apache 2.0
+// license that can be found in the LICENSE file.
+
+package webtest
+
+import (
+	"io"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"net/url"
+	"testing"
+	"time"
+)
+
+type W struct {
+	t    *testing.T
+	host string
+
+	Client *http.Client
+}
+
+func New(t *testing.T, host string) *W {
+	return &W{
+		t:      t,
+		host:   host,
+		Client: http.DefaultClient,
+	}
+}
+
+func (w *W) WaitForNet() {
+	const retryDelay = 100 * time.Millisecond
+	deadline := time.Now().Add(30 * time.Second)
+
+	for time.Now().Before(deadline) {
+		conn, err := net.Dial("tcp", w.host)
+		if err != nil {
+			time.Sleep(retryDelay)
+			continue
+		}
+		conn.Close()
+		break
+	}
```

Comment:
shouldn't something happen if the timeout is hit without net being ready? t.fatal?